Add Breeze auth services

rick-astral-cat opened the pull request:
This commit add required packages to start developing frontend and authentication services.
This packages are loaded with npm and composer so run npm run build and composer update.
-Livewire
-Laravel Breeze
-Material Icons

crazybolillo commented on composer.json:
```diff
@@ -10,10 +10,12 @@
         "laravel/sanctum": "^3.2",
-        "laravel/tinker": "^2.8"
+        "laravel/tinker": "^2.8",
+        "livewire/livewire": "^3.0"
```
Is this required for authorization?

rick-astral-cat replied:
Laravel tinker is a command line to interact with Laravel ORM installed by Laravel Breeze. I think this utility could be helpful sometimes. 
Livewire is neccesary to react components, but we can not added yet on this PR.

crazybolillo commented on package.json:
```diff
@@ -6,8 +6,20 @@
+        "tailwindcss": "^3.1.0",
         "vite": "^4.0.0"
+    },
+    "dependencies": {
+        "bootstrap": "^5.3.1",
+        "jquery": "^3.7.0",
+        "material-icons": "^1.13.10",
+        "postcss-loader": "^7.3.3"
```
I don't think these are added by breeze, at least from what I  tested. Maybe by livewire? But I think we should do this step by step.

crazybolillo commented on resources/css/app.css:
```diff
@@ -0,0 +1,7 @@
+@import "material-icons";
+@import "bootstrap";
```
Will you try to get rid of bootstrap? From what I see at least the login page is already built, I don't think we need extra styling, so I don't see the need for bootstrap as of now :thinking:

crazybolillo reviewed (commented):
It is hard to tell what change comes from installing breeze and what changes are yours. I suggest splitting it into two commits:

- First commit with everything after running `php artisan breeze:install`
- Second commit with your modifications

Also, the goal should be clear. In this case stuff like livewire or bootstrap are added but not really used. I think this PR should only focus on making user authentication work, so any extra dependencies like bootstrap or jqueyr should not be here, it is better to add them once you add the components that need them.

crazybolillo commented:
Looks good overall, just need some organization. It will make things easier if we ever want to go back and review old code.

rick-astral-cat committed:
ADD: auth scaffold, update deps

This commit installs breeze and all the scaffold code generated to
enable authorization on the app. It also updates some other dependencies
to the newest patch version.

Code has not been formatted or touched in any way. This is code exactly
as generated by breeze.

crazybolillo committed:
REF: generate baseline for phpstan, run prettier

This commit fixes failing hooks when running pre-commit. It generates a
baseline for phpstan in order to ignore scaffold code that won't be
touched for now.

JS files were also modified by prettier to standardize them.

crazybolillo commented:
@rick-astral-cat

I have rebased it to fix the CI. If you wish you can merge this or push your commits to customize the login page here.
